File: Runtime/Plugins/iOS/Source/base/ApiResult.swift
import AffiseInternal

class ApiResult : InternalResult {
    
    private var result: Any?
    private var error: String?
    private var implemented: Bool = false
    
    func success(_ result: Any?) {
        self.result = result
    }
    
    public func error(_ error: String) {
        self.error = error
    }
        
    public func notImplemented() {
        implemented = true;
    }
    
    func getResult() -> Any? {
        return result
    }
    
    func getError() -> String? {
        return error
    }
    
    func isNotImplemented() -> Bool {
        return implemented
    }
}

File: Runtime/Plugins/unity.androidlib/src/main/java/com/affise/attribution/unity/ApiResult.java
package com.affise.attribution.unity;

import com.affise.attribution.internal.callback.InternalResult;

public class ApiResult implements InternalResult {

    private Object result;
    private String error;
    private boolean isNotImplemented = false;

    @Override
    public void success(Object result) {
        this.result = result;
    }

    @Override
    public void error(String errorMessage) {
        this.error = errorMessage;
    }

    @Override
    public void notImplemented() {
        isNotImplemented = true;
    }

    public Object getResult() {
        return result;
    }

    public String getError() {
        return error;
    }

    public boolean isError() {
        return error != null;
    }

    public boolean isNotImplemented() {
        return isNotImplemented;
    }
}
